### DISCUSSION TITLE
Getting Twitch Ads again as Turbo User

### DISCUSSION THREAD

**Discussion by basti311** [streamlink/streamlink#6222] [Q&A]
### Description

Until yesterday, I was successfully able to avoid Twitch ads completely by using my OAuth token as a Twitch Turbo user. However, since exactly 2024-10-06 01:43:09 CET, it stopped working. Now, I get ads again. It's manageable with the --twitch-disable-ads option, but I was wondering if anyone knows what is going on or if Twitch is possibly ignoring Turbo user status now. If that's the case, I will cancel my membership.

I have updated to the latest version, regenerated my tokens, and even created a new application, but nothing has helped.

Any comments or suggestions would be appreciated!

        command = [
            'streamlink',
            '--loglevel', 'debug',
            '--hls-live-edge', '10',
            '--hls-live-restart',
            '--stream-segment-attempts', '10',
            '--stream-segment-timeout', '30',
            '--retry-streams', '1',
            '--retry-max', '1000',
            '--retry-open', '50',
            '--twitch-disable-ads',  
            f'--twitch-api-header=Authorization=OAuth {oauth_token}',
            f'--twitch-api-header=Client-ID={api_client_id}',
            self.stream,
            'best',
            '-o', output_file
        ]

[session][debug] Loading plugin: twitch
[cli][debug] OS:         Windows 10
[cli][debug] Python:     3.10.10
[cli][debug] OpenSSL:    OpenSSL 1.1.1s  1 Nov 2022
[cli][debug] Streamlink: 6.11.0
[cli][debug] Dependencies:
[cli][debug]  certifi: 2024.7.4
[cli][debug]  isodate: 0.6.1
[cli][debug]  lxml: 5.2.2
[cli][debug]  pycountry: 24.6.1
[cli][debug]  pycryptodome: 3.20.0
[cli][debug]  PySocks: 1.7.1
[cli][debug]  requests: 2.32.3
[cli][debug]  trio: 0.26.0
[cli][debug]  trio-websocket: 0.11.1
[cli][debug]  typing-extensions: 4.12.2
[cli][debug]  urllib3: 2.2.2
[cli][debug]  websocket-client: 1.8.0
[cli][debug]  exceptiongroup: 1.2.2
[cli][debug] Arguments:
[cli][debug]  url=https://www.twitch.tv/echteralphakevin
[cli][debug]  stream=['best']
[cli][debug]  --loglevel=debug
[cli][debug]  --output=F:\twitch streams\echteralphakevin_stream3_2024-10-06\echteralphakevin.mp4
[cli][debug]  --retry-streams=1.0
[cli][debug]  --retry-max=1000
[cli][debug]  --retry-open=50
[cli][debug]  --stream-segment-attempts=10
[cli][debug]  --stream-segment-timeout=30.0
[cli][debug]  --hls-live-edge=10
[cli][debug]  --hls-live-restart=True
[cli][debug]  --twitch-disable-ads=True
[cli][debug]  --twitch-api-header=[('Authorization', 'OAuth et7wj0w0tyccxp0jl9bthz3zhlu16a'), ('Client-ID', 'gp762nuuoqcoxypju8c569th9wz7q5')]
[cli][info] Found matching plugin twitch for URL https://www.twitch.tv/echteralphakevin
[plugins.twitch][debug] Getting live HLS streams for echteralphakevin
[plugins.twitch][debug] {'adblock': False, 'geoblock_reason': '', 'hide_ads': False, 'server_ads': True, 'show_ads': True}
[utils.l10n][debug] Language code: de_DE
[cli][info] Available streams: audio_only, 160p (worst), 360p, 480p, 720p60, 1080p60 (best)
[cli][info] Opening stream: 1080p60 (hls)
[cli][info] Writing output to
F:\twitch streams\echteralphakevin_stream3_2024-10-06\echteralphakevin.mp4
[cli][debug] Checking file output
[plugins.twitch][info] Will skip ad segments
[stream.hls][debug] Reloading playlist
[cli][debug] Pre-buffering 8192 bytes
[stream.hls][debug] First Sequence: 993; Last Sequence: 1006
[stream.hls][debug] Start offset: 0; Duration: None; Start Sequence: 993; End Sequence: None
[stream.hls][debug] Adding segment 993 to queue
[stream.hls][debug] Adding segment 994 to queue
[stream.hls][debug] Adding segment 995 to queue
[stream.hls][debug] Adding segment 996 to queue
[stream.hls][debug] Adding segment 997 to queue
[stream.hls][debug] Adding segment 998 to queue
[stream.hls][debug] Adding segment 999 to queue
[stream.hls][debug] Adding segment 1000 to queue
[stream.hls][debug] Adding segment 1001 to queue
[stream.hls][debug] Adding segment 1002 to queue
[stream.hls][debug] Adding segment 1003 to queue
[stream.hls][debug] Adding segment 1004 to queue
[stream.hls][debug] Adding segment 1005 to queue
[stream.hls][debug] Adding segment 1006 to queue
[stream.hls][debug] Writing segment 993 to output
[stream.hls][debug] Segment 993 complete
[cli][debug] Writing stream to output
2024-10-06 20:25:03,100 - INFO - Video file created for echteralphakevin at 2024-10-06 20:25:03.029620+02:00

**Comment by basti311** [comment]
Just saw that I included my Tokens, already revoked them.

**Comment by bastimeyer** [comment]
> and even created a new application
> f'--twitch-api-header=Client-ID={api_client_id}',

This doesn't work and never has. Authentication only works with Twitch's own client ID on their private GQL API that's used on their website. This has NOTHING to do with their helix REST API where you need to create your own client IDs.

What you're doing here doesn't make any sense. You've also included the OAuth token of your own account, which you should revoke, or everyone will have access to your account with the according OAuth access scopes.

Read this
https://streamlink.github.io/cli/plugins/twitch.html

**Reply by bastimeyer** [reply]
The GQL API Client ID is hardcoded for a reason:
https://github.com/streamlink/streamlink/blob/20a58b9c88e1348f453cc4bb3567e0a937ed74a2/src/streamlink/plugins/twitch.py#L302

**Reply by basti311** [reply]
Thank you for the answer. I’m not sure what is happening, as I’m not a professional programmer. All I can say is that I never saw a single ad since authenticating with my Turbo user token, and I’ve recorded around 4TB of streams.

I also didn’t use the --twitch-disable-ads option before.
